### PR TITLE
Generate SBOM only for release builds

### PR DESCRIPTION
**Pull request by Francisco-Gamino**
### Issue describing the changes in this PR

Generate SBOM only for release builds.

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -23,12 +23,28 @@ variables:
...
       $ErrorActionPreference = "Stop"
-      ./build.ps1 -Clean -Configuration Release -BuildNumber "$(buildNumber)" -AddSBOM -SBOMUtilSASUrl $env:SBOMUtilSASUrl
+      if ($isReleaseBuild)
+      {
+        ./build.ps1 -Clean -Configuration Release -BuildNumber "$(buildNumber)" -AddSBOM -SBOMUtilSASUrl $env:SBOMUtilSASUrl
+      }
+      else
+      {
+        ./build.ps1 -Clean -Configuration Release -BuildNumber "$(buildNumber)"
+      }
```
Comment (davidmrdavid) [inline azure-pipelines.yml:47]: non-blocking nit: Given the fact that the command we're running here shares the same prefix in both branches, I wonder if you could dynamically construct this command instead.

For example, I'm thinking of something like in the following **pseudocode**.
```
command <- '"./build.ps1 -Clean -Configuration Release -BuildNumber "$(buildNumber)"'
if ($isReleaseBuild)
{
  // add SBOM step
  command <- command  + "-AddSBOM -SBOMUtilSASUrl $env:SBOMUtilSASUrl"
}

Invoke(command);
```

That said, I realize this is just a configuration file so perhaps this is an unnecessary refactoring, so I'm leaving this as a non-blocking suggestion :)

**Review by davidmrdavid — APPROVED** [review]
LGTM!

File: azure-pipelines.yml (hunk 1 above)
Comment (Francisco-Gamino) [reply azure-pipelines.yml:47]: Thanks @davidmrdavid this is a good point. However, I think calling the script directly with the required parameters for each scenario is easier to read and maintain. For example, to debug a scenario where we build a command, we might have to add additional log statements to see what command is ran. If it is ok with you, I will keep it as is.

Comment (davidmrdavid) [reply azure-pipelines.yml:47]: Seems reasonable, merge ahead!

**Pull request merged**